kmir/src/kmir/kast.py

```diff
@@ -0,0 +1,210 @@
+                                'Operand::Constant',
+                                (
+                                    KApply(
+                                        'constOperand(_,_,_)_BODY_ConstOperand_Span_MaybeUserTypeAnnotationIndex_MirConst',
+                                        (
+                                            KApply('span', token(0)),
+                                            KApply('noUserTypeAnnotationIndex_BODY_MaybeUserTypeAnnotationIndex', ()),
+                                            KApply(
+                                                'mirConst(_,_,_)_TYPES_MirConst_ConstantKind_Ty_MirConstId',
+                                                (
+                                                    KApply('ConstantKind::ZeroSized', ()),
+                                                    KApply('ty', (token(target),)),
+                                                    KApply('mirConstId(_)_TYPES_MirConstId_Int', (token(0),)),
+                                                ),
+                                            ),
+                                        ),
+                                    ),
+                                ),
+                            ),
+                            args,
+                            KApply(
+                                'place',
+                                (
+                                    KApply('local', (token(0),)),
+                                    KApply('ProjectionElems::empty', ()),
+                                ),
+                            ),
+                            KApply('noBasicBlockIdx_BODY_MaybeBasicBlockIdx', ()),
+                            KApply('UnwindAction::Continue', ()),
+                        ),
+                    ),
+                    KApply('span', token(0)),
+                ),
+            ),
+        ),
+    )
+
+
+def symbolic_locals(smir_info: SMIRInfo, local_types: list[dict]) -> tuple[list[KInner], list[KInner]]:
+    locals, constraints = ArgGenerator(smir_info).run(local_types)
+    local0 = KApply('newLocal', (KApply('ty', (token(0),)), KApply('Mutability::Not', ())))
+    return ([local0] + locals, constraints)
+
+
+class ArgGenerator:
+    smir_info: SMIRInfo
+    locals: list[KInner]
+    pointees: list[KInner]
+    constraints: list[KInner]
+    counter: int
+    ref_offset: int
+
+    if TYPE_CHECKING:
+        from .smir import Ty
+
+    def __init__(self, smir_info: SMIRInfo) -> None:
+        self.smir_info = smir_info
+        self.locals = []
+        self.pointees = []
+        self.constraints = []
+        self.counter = 1
+        self.ref_offset = 0
+
+    def run(self, local_types: list[dict]) -> tuple[list[KInner], list[KInner]]:
+        self.ref_offset = len(local_types) + 1
+        for ty in [t['ty'] for t in local_types]:
+            self._add_local(ty)
+        return (self.locals + self.pointees, self.constraints)
```

Comment:
If you call this method `def __call__(...)`, you can do `ArgGenerator(smir_info)(local_types)` directly. See https://github.com/runtimeverification/k/blob/db2706d64499fda526e2a7b374bf24c32436e7c3/pyk/src/pyk/cterm/cterm.py#L354 for an example. Stylistic choice.